rubrics/tool/src/main/java/org/sakaiproject/rubrics/controller/GeneratePdfController.java

```diff
@@ -0,0 +1,263 @@
+import com.itextpdf.text.pdf.PdfWriter;
+
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.user.api.PreferencesService;
+
+import org.jsoup.Jsoup;
+
+@Slf4j
+@RepositoryRestController
+@NoArgsConstructor
+@AllArgsConstructor
+public class GeneratePdfController {
+
+    @Autowired
+    RubricsConfiguration rubricsConfiguration;
+
+    @Autowired
+    RepositoryEntityLinks repositoryEntityLinks;
+    
+    @Autowired
+    private MessageSource messageSource;
+    
+    @Autowired
+    private SiteService siteService;
+    
+    @Autowired
+    private RubricRepository rubricRepository;
+    
+    @Autowired
+    private EvaluationRepository evaluationRepository;
+    
+    @Autowired
+    private SessionManager sessionManager;
+
+    @Autowired
+    private PreferencesService preferencesService;
+
+    private static final String X_COPY_HEADER = "x-copy-source";
+    private static final String X_LANG_HEADER = "lang";
+    private static Font font = new Font(Font.FontFamily.HELVETICA, 10, Font.BOLD);
+    private static Font textFont = new Font(Font.FontFamily.HELVETICA, 8, Font.BOLD);
+    private static Font descFont = new Font(Font.FontFamily.HELVETICA, 7, Font.NORMAL);
+
+    @RequestMapping(value = "/rubrics/generatePdf", headers = {X_COPY_HEADER,X_LANG_HEADER}) 
+    public ResponseEntity<StreamingResponseBody> generatePdf(@Param("sourceId") @RequestHeader(value = X_COPY_HEADER) String sourceId, 
+            @RequestHeader(value = X_LANG_HEADER) String lang, @RequestHeader(value = "toolId") String toolId,
+            @RequestHeader(value = "itemId") String entityId,
+            @RequestHeader(value = "evaluatedItemId") String evaluatedItemId)
+            throws Exception {
+
+        Rubric sourceRubric = null;
+        String fileRubricName = null;
+        List<Evaluation> evals = null;
+        StreamingResponseBody stream = null;
+
+        try {
+            sourceRubric = rubricRepository.findById(Long.parseLong(sourceId))
+                .orElseGet(() -> rubricsConfiguration.getDefaultLayoutConfiguration(lang).getDefaultRubric());
+
+            evals = evaluationRepository.findByToolIdAndAssociationItemIdAndEvaluatedItemId(toolId, entityId, evaluatedItemId);
+            String fileName = sourceRubric.getTitle();
+            fileRubricName = fileName + ".pdf";
+            File tempFile = createPdf(sourceRubric, fileRubricName, evals);
+            Path pdfPath = Paths.get(tempFile.getAbsolutePath());
+            stream = outputStream -> {
+                Files.copy(pdfPath, outputStream);
+            };
+            //Files.deleteIfExists(pdfPath);
+        } catch (DocumentException | IOException ex) {
+            log.error(ex.getMessage(), ex);
+        }
+
+        HttpHeaders respHeaders = new HttpHeaders();
+        respHeaders.setContentType(new MediaType("application", "pdf"));
```

Comment:
If you use `produces = MediaType.APPLICATION_PDF_VALUE` in the RequestMapping, I believe it will eliminate the need to specify the headers here.